lib/src/platform/src/realtime/connection.dart

```diff
@@ -21,9 +25,11 @@ class Connection extends PlatformObject {
   Connection(this.realtime)
       : _state = ConnectionState.initialized,
         super() {
-    on().listen((event) {
-      _state = event.current;
-      _errorReason = event.reason;
+    _onConnectionStateChange().listen((event) {
+      _state = event.stateChange.current;
+      _errorReason = event.stateChange.reason;
+      _id = event.connectionId;
```

Comment:
I think it should be `_id = this.realtime.connectionId` where `connectionId` is something like [this](https://github.com/ably/ably-flutter/blob/2e1839d609ca076ff6e7c3e4f039c818c3bec25d/android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java#L579), but not async, wdyt @ttypic @sacOO7 ?

Reply:
`private void connectionId(@NonNull MethodCall methodCall, @NonNull MethodChannel.Result result)` can't be called from dart code? via `invoke` or something

Reply:
We can connect over DM